app/screens/send-bitcoin-screen/send-lnurl-screen.tsx

```diff
@@ -0,0 +1,323 @@
+  const [maxMemoSize, setMaxMemoSize] = useState(0)
+
+  const [queryTransactions] = useLazyQuery(QUERY_TRANSACTIONS, {
+    fetchPolicy: "network-only",
+  })
+
+  const [payInvoice] = useMutation(PAY_INVOICE, {
+    update: () => queryTransactions(),
+  })
+
+  const hasEnoughBalance = balance >= amount
+  const isAmountSendable = isAmountValid(amount, minSendable, maxSendable)
+
+  useEffect(() => {
+    const parseInitialInvoice = async () => {
+      const lnurl = await parseUrl(route.params.invoice)
+      const { tag, callback, minSendable, maxSendable, commentAllowed, metadata } = lnurl
+
+      if (!isProtocolSupported(tag)) {
+        Alert.alert("LNUrl protocol not supported")
+        throw new Error("LNURL protocol not supported")
+      }
+
+      setAmount(mSatToSat(minSendable))
+      setPrimaryAmountValue(mSatToSat(minSendable))
+      setSecondaryAmountValue(mSatToSat(minSendable))
+      setCallback(callback)
+      setMinSendable(mSatToSat(minSendable))
+      setMaxSendable(mSatToSat(maxSendable))
+      setMaxMemoSize(commentAllowed)
+
+      try {
+        const description = JSON.parse(metadata)[0][1]
+        setDescription(description)
+      } catch (error) {
+        setDescription("Invoice description not recognized")
+      }
+    }
+
+    parseInitialInvoice()
+  }, [route.params.invoice, setPrimaryAmountValue, setSecondaryAmountValue])
+
+  const submitPayment = async () => {
+    setSubmitStatus(1)
+
+    try {
+      const paymentInvoice = await invoiceRequest(callback, amount, memo)
+      const payment = await payInvoice({
+        variables: { invoice: paymentInvoice, memo: description },
+      })
+      const { data } = payment
+
+      setSubmitStatus(0)
+
+      if (
+        data.invoice.payInvoice === "success" ||
+        data.invoice.payInvoice === "pending"
+      ) {
+        setStatus(PAYMENT_STATUS.Success)
+      } else {
+        setStatus(PAYMENT_STATUS.Error)
+      }
+    } catch (error) {
+      setSubmitStatus(0)
+      setStatus(PAYMENT_STATUS.Error)
+    } finally {
+      setTimeout(() => {
+        setStatus(PAYMENT_STATUS.Finished)
+      }, 4000)
+    }
+  }
+
+  return (
+    <Screen preset="fixed">
+      <ScrollView style={styles.mainView} contentContainerStyle={styles.scroll}>
+        <View style={styles.section}>
+          <InputPayment
+            editable={true}
+            onUpdateAmount={(input) => {
+              setAmount(input)
+            }}
+            forceKeyboard
+            price={btcPrice}
+            primaryAmount={primaryAmount}
+            secondaryAmount={secondaryAmount}
+          />
+
+          <View style={styles.errorContainer}>
+            {!hasEnoughBalance && <Text>You do not have enough sats</Text>}
+          </View>
+        </View>
+        <View style={styles.transactionDetailView}>
+          <Text>
+            Please pay between{" "}
+            {minSendable === null ? "Loading..." : `${minSendable.toString()} sats`}
```

Comment:
I think it would be better to just have an ActivityIndicator if the lnurl data is still loading.